pkg/controller/build/build_controller.go

```diff
@@ -204,7 +273,12 @@ func (r *ReconcileBuild) Reconcile(request reconcile.Request) (reconcile.Result,
 
 func (r *ReconcileBuild) validateRuntime(runtime *build.Runtime) error {
 	if len(runtime.Paths) == 0 {
-		return fmt.Errorf("the property 'spec.runtime.paths' must not be empty")
+		errorCode, err := ValidateAndGenerateBuildErrorCode(RuntimePathsCanNotBeEmpty)
```

Comment:
`ValidateAndGenerateBuildErrorCode` should not happen here. Same applies to all other places in code where this is done. I think in this scenarios, we just wanna return the error containing the right code, e.g:

```
return fmt.Errorf(errorCode + ": " + "the property 'spec.runtime.paths' must not be empty")
```
or
```
 return errors.Errorf("%v: secrets %s do not exist", SecretsDoNotExist, missingSecrets)
```

Reply:
The validation should be part of unit tests, in a way that we can detect:
- if an errorcode was deleted

I dont think it hurts to add new validations, and basically an existing one cannot be modified because the definition comes from IOTA. I think what I´m trying to say is that `ListSecretInNamespaceFailed` can be modified to `ListSecretInNamespaceFailedFoobar` but the underlying code will remain the same, basically `2`. Then via unit-tests, we should validate that for every validation the `Status.Reason` contains the right error code.